python/plugins/processing/algs/gdal/rasterize_over.py

```diff
@@ -125,3 +125,25 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append(inLayer.source())
 
         return [self.commandName(), GdalUtils.escapeAndJoin(arguments)]
+
+    def postProcessAlgorithm(self, context, feedback):
+        fileName = self.output_values.get(self.OUTPUT)
+        if not fileName:
+            return {}
+
+        if context.project():
+            for l in context.project().mapLayers().values():
+                if l.source() != fileName:
+                    continue
+
+                l.dataProvider().reloadData()
+                l.triggerRepaint()
+
+        for l in context.temporaryLayerStore().mapLayers().values():
+            if l.source() != fileName:
+                continue
+
+            l.dataProvider().reloadData()
+            context.temporaryLayerStore().addMapLayer(l)
```

Comment:
why do we re-add this here? won't it already be in that store?

Reply:
It is re-added in order to have an updated layer with the reloaded data. But seems you are right and this line can be safely removed.

Reply:
Thanks! Ultimately I think we should consider moving this logic to a function in QgsProcessingContext (something like ::reloadLayersWithPath(....) ), but that's entirely optional 👍

Reply:
Actually maybe ::invalidateLayersAtPath(...) is better

Reply:
Do you think it should be done as a part of this PR or it is better to do it in a follow-up PR targeted only at master branch?

Reply:
Yep, a follow up PR sounds good!